frontend/src/layouts/Header.tsx

```diff
@@ -10,14 +11,22 @@ export default function Header () {
   }
 
   function buttonLoginLogout() {
-    const authenticated = auth.isAuthenticated
-    if(authenticated) {
+    if(auth.isAuthenticated) {
       return <Button style={buttonStyle} onClick={() => auth.signoutRedirect()}>Log out</Button>
     } else {
       return <Button style={buttonStyle} onClick={() => auth.signinRedirect()}>Log in</Button>
     }
   }
 
+  useEffect(() => {
+    const token = auth.user?.access_token
+    if(token) {
+      localStorage.setItem('token', token)
+    } else {
+      localStorage.removeItem('token')
+    }
+  }, [auth.user?.access_token])
+
```

Comment:
Der Code wäre vielleicht in der Main Komponente oder vielleicht sogar einer eigenen Auth-Komponente um die Main Komponente etwas besser aufgehoben.